File: BOJ/♣ 22000~22999/22151_Игра.py
N = int(input())
for _ in range(N):
    limit, start_r, start_c, end_r, end_c = map(int, input().split())
    dist = abs(start_r - end_r) + abs(start_c - end_c)

    if dist <= limit:
        print('First 1')
    else:
        print('Infinity')

File: BOJ/♣ 22000~22999/22151_Игра.cpp
#include <algorithm>
#include <iostream>
using namespace std;

int main() {
    int T;
    cin >> T;

    while (T--) {
        int limit, start_r, start_c, end_r, end_c;
        cin >> limit >> start_r >> start_c >> end_r >> end_c;

        int dist = abs(start_r - end_r) + abs(start_c - end_c);

        if (dist <= limit) {
            cout << "First 1\n";
        } else {
            cout << "Infinity\n";
        }
    }
}